### PR TITLE
Support requested_reviewers data in comment webhook events

### LINKED ISSUE #25833: Provide reviewers/participants in webhook data for pull requests
### Feature Description

The json data posted by outgoing webhooks for pull request events currently contains assignees, but not reviewers or participants (both of which show in the UI as part of a PR).
It would be very useful to have a list of the reviewers and participants in the webhook data.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Makonike**
close #25833 

Currently, the information for "requested_reviewers" is only included in the webhook event for reviews. I would like to suggest adding this information to the webhook event for "PullRequest comment" as well, as they both pertain to the "PullRequest" event.

Also, The reviewer information for the Pull Request is not displayed when it is approved or rejected.

### CODE REVIEW

File: services/webhook/notifier.go [hunk 1]
```diff
@@ -376,17 +376,19 @@ func (m *webhookNotifier) NotifyUpdateComment(ctx context.Context, doer *user_mo
 	}
 
 	var eventType webhook_module.HookEventType
+	var pullRequest *api.PullRequest
 	if c.Issue.IsPull {
 		eventType = webhook_module.HookEventPullRequestComment
+		pullRequest = convert.ToAPIPullRequest(ctx, c.Issue.PullRequest, nil)
```
Comment (lunny) [inline services/webhook/notifier.go]: The third parameter is necessary which should be the `doer`.

Comment (Makonike) [reply services/webhook/notifier.go]: All fixed now. Sorry for the long wait on this PR~

File: modules/structs/hook.go [hunk 2]
```diff
@@ -217,13 +217,14 @@ const (
...
-	Action     HookIssueCommentAction `json:"action"`
-	Issue      *Issue                 `json:"issue"`
-	Comment    *Comment               `json:"comment"`
-	Changes    *ChangesPayload        `json:"changes,omitempty"`
-	Repository *Repository            `json:"repository"`
-	Sender     *User                  `json:"sender"`
-	IsPull     bool                   `json:"is_pull"`
+	Action      HookIssueCommentAction `json:"action"`
+	Issue       *Issue                 `json:"issue"`
+	PullRequest *PullRequest           `json:"pull_request"`
```
Comment (lunny) [inline modules/structs/hook.go]:
```suggestion
	PullRequest *PullRequest           `json:"pull_request,omitempty"`
```

Comment (Makonike) [reply modules/structs/hook.go]: Done.

File: services/webhook/notifier.go [hunk 3]
```diff
@@ -440,20 +446,23 @@ func (m *webhookNotifier) DeleteComment(ctx context.Context, doer *user_model.Us
 	}
 
 	var eventType webhook_module.HookEventType
+	var pullRequest *api.PullRequest
 	if comment.Issue.IsPull {
 		eventType = webhook_module.HookEventPullRequestComment
+		pullRequest = convert.ToAPIPullRequest(ctx, comment.Issue.PullRequest, nil)
```
Comment (lunny) [inline services/webhook/notifier.go]:
```suggestion
		pullRequest = convert.ToAPIPullRequest(ctx, comment.Issue.PullRequest, doer)
```

Comment (Makonike) [reply services/webhook/notifier.go]: I've noticed that many places need the doer parameter added. Should I update them all at once?

Comment (lunny) [reply services/webhook/notifier.go]: Yes, please.

Comment (Makonike) [reply services/webhook/notifier.go]: Done.

**Review by lunny — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Pull request merged**